File: src/main/java/stream_homework_01/Question_03.java
package stream_homework_01;

import java.util.Arrays;
import java.util.Comparator;
import java.util.List;

/**
 * 학생 객체의 리스트가 주어졌을 때, 성적(score)이 80점 이상인 학생들만을 선택하고, 이들의 이름을 알파벳 순으로 정렬하여 출력하시오.
 */
public class Question_03 {

    public static void main(String[] args) {
        List<Student03> students = Arrays.asList(
                new Student03("Alice", 82),
                new Student03("Bob", 90),
                new Student03("Charlie", 72),
                new Student03("David", 76)
        );

        students.stream()
                .filter(student -> student.getScore() >= 80)
                .sorted(Comparator.comparing(Student03::getName))
                .forEach(student -> System.out.println(student.getName()));
    }
}

class Student03 {
    private String name;
    private int score;

    public Student03(String name, int score) {
        this.name = name;
        this.score = score;
    }

    public String getName() {
        return name;
    }

    public int getScore() {
        return score;
    }
}

File: HomeworkJS/0923_js/question_03.js
// 학생 정보를 담은 객체를 만드세요
// 포함할 정보: 이름, 나이, 학년, 좋아하는 과목들(배열)

// 그리고 다음 작업들을 해보세요:
// 1. 학생의 이름을 출력하기
// 2. 학생의 나이를 1살 증가시키기
// 3. 새로운 과목을 좋아하는 과목 배열에 추가하기
// 4. 학생 정보를 모두 출력하기

const student = {
  name: "홍길동",
  age: 20,
  grade: 1,
  favoriteSubject: ["체육", "수학"],
};

console.log(student.name);

student.age++;

student.favoriteSubject.push("음악");

console.log(student);
